src/util.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Context as _;
+use futures::{future::BoxFuture, FutureExt as _, TryStreamExt as _};
+use std::path::PathBuf;
+use tokio::fs;
+
+pub fn copy_dir(
+    src: impl Into<PathBuf>,
+    dst: impl Into<PathBuf>,
+) -> BoxFuture<'static, anyhow::Result<()>> {
+    copy_dir_impl(src.into(), dst.into()).boxed()
+}
+
+async fn copy_dir_impl(src: PathBuf, dst: PathBuf) -> anyhow::Result<()> {
```

Comment:
There's like million lines of `with_context` in this function, but debugging recursive copy errors is the most annoying thing ever, so I stand by this.